.github/workflows/parachain.yml

```diff
@@ -16,116 +16,36 @@ on:
   workflow_dispatch:
 
 jobs:
-  check:
+  build:
     runs-on: snowbridge-runner
     env:
       CARGO_INCREMENTAL: 0
       RUST_BACKTRACE: 1
       RUSTFLAGS: -C debuginfo=1
       SKIP_WASM_BUILD: 1
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v1
-        with:
-          path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-          key: ${{ runner.os }}-cargo-${{ hashFiles('parachain/Cargo.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-cargo-
-      - name: install protoc
-        uses: arduino/setup-protoc@v1
-        with:
-          repo-token: ${{ secrets.GITHUB_TOKEN }}
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: nightly-2023-04-22
-          target: wasm32-unknown-unknown
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/install_toolchain
       - name: cargo check
-        uses: actions-rs/cargo@v1
-        with:
-          command: check
-          toolchain: nightly-2023-04-22
-          args: >-
-            --manifest-path parachain/Cargo.toml
-            --workspace
-            --features runtime-benchmarks
-  test:
-    needs: check
-    runs-on: snowbridge-runner
-    env:
-      CARGO_INCREMENTAL: 0
-      RUST_BACKTRACE: 1
-      RUSTFLAGS: -C debuginfo=1
-      SKIP_WASM_BUILD: 1
-    steps:
-      - uses: actions/checkout@v2
-        with:
-          ref: ${{ github.head_ref }}
-      - uses: actions/cache@v1
-        with:
-          path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-          key: ${{ runner.os }}-cargo-${{ hashFiles('parachain/Cargo.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-cargo-
-      - uses: arduino/setup-protoc@v1
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: nightly-2023-04-22
-          target: wasm32-unknown-unknown
-      # Increase stack limit for beacon light client tests
-      - run: sudo prlimit --pid $$ --stack=32768
```

Comment:
I needed this for the previous rust compiler version. I would leave it in just in case future rust versions cause problems.